include/util/hex_float.h

```diff
@@ -213,12 +270,310 @@ class HexFloat {
   static const uint32_t fraction_right_shift =
       (sizeof(uint_type) * 8) - num_fraction_bits;
 
+  // The maximum representable normalized exponent.
+  static const int_type max_exponent =
+      (exponent_mask >> num_fraction_bits) - exponent_bias;
+  // The minimum representable exponent for normalized numbers.
+  static const int_type min_exponent = -exponent_bias;
```

Comment:
Min biased exponent for normal or subnormal numbers is 1, so 1-15 = -14 is min exponent.  Not -15.
Or maybe you need a different comment?